src/main/java/com/minecolonies/coremod/client/gui/WindowHomeBuilding.java

```diff
@@ -1,24 +1,147 @@
 
 /**
  * Window for the home building.
  */
 public class WindowHomeBuilding extends AbstractWindowBuilding<BuildingHome.View>
 {
+    /**
+     * Id of the hire/fire button in the GUI.
+     */
+    private static final String BUTTON_ASSIGN = "assign";
+
+    /**
+     * Id of the hire/fire button in the GUI.
+     */
+    private static final String BUTTON_REMOVE = "remove";
+
+    /**
+     * Suffix describing the window xml.
+     */
     private static final String HOME_BUILDING_RESOURCE_SUFFIX = ":gui/windowHutHome.xml";
 
     /**
+     * Id to identify the list of the citizen in the view.
+     */
+    private static final String LIST_CITIZEN = "assignedCitizen";
+
+    /**
+     * The list of citizen assigned to this hut.
+     */
+    private ScrollingList citizen;
+
+    /**
+     * The building the view is relates to.
+     */
+    private final BuildingHome.View home;
+
+    /**
      * Creates the Window object.
      *
      * @param building View of the home building.
      */
     public WindowHomeBuilding(final BuildingHome.View building)
     {
         super(building, Constants.MOD_ID + HOME_BUILDING_RESOURCE_SUFFIX);
+
+        super.registerButton(BUTTON_ASSIGN, this::assignClicked);
+        super.registerButton(BUTTON_REMOVE, this::removeClicked);
+        this.home = building;
+    }
+
+    /**
+     * Refresh the view.
+     */
+    private void refreshView()
+    {
+        final Button buttonAssign = findPaneOfTypeByID(BUTTON_ASSIGN, Button.class);
+
+        final int sparePlaces = building.getBuildingLevel() - building.getResidents().size();
+        buttonAssign.setLabel(LanguageHandler.format(COM_MINECOLONIES_COREMOD_GUI_HOME_ASSIGN, sparePlaces));
+        buttonAssign.setEnabled(sparePlaces > 0 && building.getColony().isManualHousing());
+
+        citizen.refreshElementPanes();
+    }
+
+    @Override
+    public void onOpened()
+    {
+        super.onOpened();
+        final boolean isManualHousing = building.getColony().isManualHousing();
+        citizen = findPaneOfTypeByID(LIST_CITIZEN, ScrollingList.class);
+        citizen.setDataProvider(new ScrollingList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return home.getResidents().size();
+            }
+
+            @Override
+            public void updateElement(final int index, @NotNull final Pane rowPane)
+            {
+                rowPane.findPaneOfTypeByID("name", Label.class).setLabelText(home.getColony().getCitizen(home.getResidents().get(index)).getName());
+                rowPane.findPaneOfTypeByID(BUTTON_REMOVE, Button.class).setEnabled(isManualHousing);
+            }
+        });
+
+        refreshView();
+    }
+
+    /**
+     * Action when an assign button is clicked.
+     *
+     * @param button the clicked button.
+     */
+    private void assignClicked(@NotNull final Button button)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "button". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1172)